### PR TITLE
Welling/replace metadata assay type improvements

### PR DESCRIPTION
**Pull request by jswelling**
This PR involves improvements to a utility script used to substitute replacement assay type strings into metadata.tsv files.  The changes relate to preserving permissions when files are moved or duplicated, and coping with assay type strings containing spaces.

### CODE REVIEW

**Review by Daghis — CHANGES_REQUESTED** [review]

File: src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh [hunk 1]
```diff
@@ -1,7 +1,11 @@
 #!/bin/bash -ex
 
-old_assay_name='MALDI-IMS-neg'
-new_assay_name='MALDI-IMS'
+set -e  # really exit on error
+
+#old_assay_name='LC-MS (metabolomics)'
```
[the remark this answers is not in the document]
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh:5]: That's fair- they are previous cases in which I actually employed the script, left for my reference, but you are correct that that is bad practice.

[the remark this answers is not in the document]
Comment (Daghis) [reply src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh:5]: I hate to call it out as "bad practice", but it's something I picked up from Google (no unused code) that seemed worth keeping in mind.

File: src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh [hunk 2]
```diff
@@ -13,6 +17,21 @@ fi
...
+dir_prot=`stat -c '%a' .`
+chmod u+w .
 mkdir -p extras
+extras_prot=`stat -c '%a' extras`
+chmod u+w extras
+prot=`stat -c '%a' $fname`
 mv $fname extras/${fname}.orig
-sed "s/${old_assay_name}/${new_assay_name}/" < extras/${fname}.orig > $fname
+ESCAPED_REPLACE=$(printf '%s\n' "$new_assay_name" | sed -e 's/[\/&]/\\&/g')
+ESCAPED_KEYWORD=$(printf '%s\n' "$old_assay_name" | sed -e 's/[]\/$*.^[]/\\&/g');
```
[the remark this answers is not in the document]
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh:28]: Nope. Fixed.

File: src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh [hunk 3]
```diff
@@ -13,6 +17,21 @@ fi
 fname=`ls *metadata.tsv | head -1`
 echo $fname
 
+dir_prot=`stat -c '%a' .`
```
[the remark this answers is not in the document]
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh:20]: fixed.

File: src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh [hunk 4]
```diff
@@ -13,6 +17,21 @@ fi
...
+chmod u+w .
 mkdir -p extras
+extras_prot=`stat -c '%a' extras`
```
[the remark this answers is not in the document]
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh:23]: fixed.

File: src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh [hunk 5]
```diff
@@ -13,6 +17,21 @@ fi
...
+chmod u+w extras
+prot=`stat -c '%a' $fname`
```
[the remark this answers is not in the document]
Comment (jswelling) [reply src/ingest-pipeline/misc/tools/replace_metadata_assay_type.sh:25]: fixed

**Pull request merged**